daemon/image_delete.go

```diff
@@ -160,17 +171,23 @@ func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
 			return err
 		}
 
+		topLevel := true
 		if err := daemon.graph.WalkHistory(parent, func(p image.Image) error {
 			if imgID == p.ID {
-				if container.IsRunning() {
-					if force {
-						return fmt.Errorf("Conflict, cannot force delete %s because the running container %s is using it, stop it and retry", stringid.TruncateID(imgID), stringid.TruncateID(container.ID))
+				if topLevel {
+					if container.IsRunning() {
```

Comment:
Please fix the style here. I can't even tell what's going on. Use exit first style:

```
if condition {
    return fmt.Errorf(...)
}

// ... do things here
```

Alternatively, you are skipping some record, use a continue:

```
if condition {
    continue // do not process record for some reason
}

// ... continue processing that record.
```